esx_service/utils/auth_api.py

```diff
@@ -350,41 +410,80 @@ def _tenant_vm_ls(name):
         return error_info, None
     
     if not tenant:
-        error_info = error_code.TENANT_NOT_EXIST.format(name)
+        error_info = error_code.generate_error_info(ErrorCode.TENANT_NOT_EXIST, name)              
         return error_info, None
     # tenant.vms is a list of vm_uuid of vms which belong to this tenant
     return None, tenant.vms
+
+def _tenant_vm_replace(name, vm_list):
+    """ API to replace vms for a tenant """
+    logging.debug("_tenant_vm_replace: name=%s vm_list=%s", name, vm_list)
```

Comment:
Should check vm_list if its none and then use for logging. Or log inside the "if" block below.